include/sys/util.h

```diff
@@ -178,6 +178,16 @@ extern "C" {
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #endif
 
+/**
+ * @def CLAMP
+ * @brief Clamp a value to a given range.
+ * @note Arguments are evaluated multiple times.
+ */
+#ifndef CLAMP
+/* Use Z_CLAMP for a GCC-only, single evaluation version */
+#define CLAMP(val, low, high) MIN(MAX(val, low), high)
```

Comment:
Wouldn't the following definition:
```
#define CLAMP(val, low, high) (((val) <= (low)) ? (low) : MIN(val, high))
```
save at least one comparison when `val` is not greater than `low`?

Reply:
Yes. We'll deviate from the Linux kernel definition, but it'll be a pretty trivial difference.